docker-compose.yml

```diff
@@ -62,6 +62,14 @@ x-ccache: &ccache
   CCACHE_MAXSIZE: 500M
   CCACHE_DIR: /ccache
 
+# CPU/memory limits to pass to Docker. These values emulate GitHub Actions:
+# https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+x-limits:
+  # Note we use cpuset and not cpus since Ninja only detects and limits parallelism
```

Comment:
Tested `--cpus` and `--cpuset-cpus`.
With `--cpuset-cpus=0,1`, ninja spawns 2 jobs, as expected.
With `--cpus=2`, ninja spawns 32 jobs on my 32 core machine, though only 2 host cores can be used.

Reply:
For reference, ninja explicitly checks for cpuset: https://github.com/ninja-build/ninja/commit/1bcc689324bdee090eed035353724abc3fa7c909

Reply:
If I understand correctly we can't set the limits manually, just use preset(s) defined in the compose file?
If so we may want to define multiple presets in the future, e.g. big-memory, small-memory, travis, etc. 

How about a `--resource-limit=github` or `--limit-preset=github` or something similar on the CLI and a mapping under `x-limit-presets` in the compose file?

Reply:
Thanks - updated it to allow defining multiple presets, though there's still only the GitHub one defined by default.